source/expectations/matchers/ut_equal.tpb

```diff
@@ -194,15 +203,29 @@ create or replace type body ut_equal as
     l_result ut_equal := self;
   begin
     l_result.is_unordered := ut_utils.boolean_to_int(true);
-    ut_utils.append_to_list(l_result.join_columns, a_columns);
+    l_result.join_columns := l_result.join_columns multiset union all coalesce(ut_utils.string_to_table(REPLACE(a_columns,'|',','),','),ut_varchar2_list());
+    
+    select regexp_replace(column_value,'^((/ROW/)|^(//)|^(/\*/))?(.*)','\5') col_names
+    bulk collect into l_result.join_on_list
+    from table(l_result.join_columns);
     return l_result;
   end;
 
   member function join_by(a_columns ut_varchar2_list) return ut_equal is
     l_result ut_equal := self;
+    l_items  ut_varchar2_list := ut_varchar2_list();
   begin
     l_result.is_unordered := ut_utils.boolean_to_int(true);
-    l_result.join_columns := l_result.join_columns multiset union all coalesce(a_columns,ut_varchar2_list());
+    for i in 1..a_columns.count loop
+      --TODO :  idoiot proof solution for both include and exclude
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this "TODO" comment. ([plsql:S1135](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=801&issues=AWd_NeZHCh2zBQVNm3D2&open=AWd_NeZHCh2zBQVNm3D2)

[ISSUE_KEY]: AWd_NeZHCh2zBQVNm3D2